e2e-tests/insurance/cypress/e2e/journeys/your-buyer/outstanding-or-overdue-payments/outstanding-or-overdue-payments.spec.js

```diff
@@ -0,0 +1,130 @@
+    });
+  });
+
+  beforeEach(() => {
+    cy.saveSession();
+  });
+
+  after(() => {
+    cy.deleteApplication(referenceNumber);
+  });
+
+  it('renders core page elements', () => {
+    cy.corePageChecks({
+      pageTitle: CONTENT_STRINGS.PAGE_TITLE,
+      currentHref: `${ROOT}/${referenceNumber}${OUTSTANDING_OR_OVERDUE_PAYMENTS}`,
+      backLink: `${ROOT}/${referenceNumber}${CURRENCY_OF_LATE_PAYMENTS}`,
+    });
+  });
+
+  describe('page tests', () => {
+    beforeEach(() => {
+      cy.navigateToUrl(url);
+    });
+
+    it('renders a heading caption', () => {
+      cy.checkText(headingCaption(), CONTENT_STRINGS.HEADING_CAPTION);
+    });
+
+    describe(TOTAL_OUTSTANDING_PAYMENTS, () => {
+      it(`should render a label for ${TOTAL_OUTSTANDING_PAYMENTS}`, () => {
+        cy.checkText(field(TOTAL_OUTSTANDING_PAYMENTS).label(), FIELDS[TOTAL_OUTSTANDING_PAYMENTS].LABEL);
+      });
+
+      it(`should render an input for ${TOTAL_OUTSTANDING_PAYMENTS}`, () => {
+        field(TOTAL_OUTSTANDING_PAYMENTS).input().should('be.visible');
+      });
+    });
+
+    describe(TOTAL_AMOUNT_OVERDUE, () => {
+      it(`should render a label for ${TOTAL_AMOUNT_OVERDUE}`, () => {
+        cy.checkText(field(TOTAL_AMOUNT_OVERDUE).label(), FIELDS[TOTAL_AMOUNT_OVERDUE].LABEL);
+      });
+
+      it(`should render an input for ${TOTAL_AMOUNT_OVERDUE}`, () => {
+        field(TOTAL_AMOUNT_OVERDUE).input().should('be.visible');
+      });
+    });
+  });
+
+  describe('form submission', () => {
+    describe('when submitting a fully filled form', () => {
+      it(`should redirect to ${FAILED_TO_PAY} page`, () => {
+        cy.navigateToUrl(url);
+
+        cy.completeAndSubmitOutstandingOrOverduePaymentsForm({});
+
+        cy.assertUrl(failedToPayUrl);
+      });
+
+      describe('when going back to the page', () => {
+        it('should have the submitted values', () => {
+          cy.navigateToUrl(url);
+
+          cy.checkValue(field(TOTAL_OUTSTANDING_PAYMENTS), BUYER[TOTAL_OUTSTANDING_PAYMENTS]);
+          cy.checkValue(field(TOTAL_AMOUNT_OVERDUE), BUYER[TOTAL_AMOUNT_OVERDUE]);
+        });
+      });
+
+      describe(`changing ${OUTSTANDING_PAYMENTS} from "yes" to "no"`, () => {
+        beforeEach(() => {
+          cy.navigateToUrl(tradingHistoryUrl);
+
+          // submit OUTSTANDING_PAYMENTS as yes
+          cy.completeAndSubmitTradingHistoryWithBuyerForm({ outstandingPayments: true });
+
+          cy.navigateToUrl(tradingHistoryUrl);
+
+          // change OUTSTANDING_PAYMENTS to no
+          cy.completeAndSubmitTradingHistoryWithBuyerForm({});
+        });
+
+        describe('when going back to the page', () => {
+          it(`should have the submitted values and have removed data from ${TOTAL_OUTSTANDING_PAYMENTS} and ${TOTAL_AMOUNT_OVERDUE}`, () => {
+            cy.navigateToUrl(url);
+
+            cy.checkValue(field(TOTAL_OUTSTANDING_PAYMENTS), '');
+            cy.checkValue(field(TOTAL_AMOUNT_OVERDUE), '');
```

Comment:
Do we have an `assertEmpty...FieldValues` type command that can be used here?

Reply:
we do not :(

Reply:
Could we create a new command for this? In `main-iterations` there's 4x instances of this :/

Reply:
Created EMS-3810 and EMS-3812